logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendTextMessageUseCase.kt

```diff
@@ -55,15 +59,16 @@ class SendTextMessageUseCase internal constructor(
     private val messageSendFailureHandler: MessageSendFailureHandler,
     private val userPropertyRepository: UserPropertyRepository,
     private val selfDeleteTimer: ObserveSelfDeletionTimerSettingsForConversationUseCase,
-    private val dispatchers: KaliumDispatcher = KaliumDispatcherImpl
+    private val dispatchers: KaliumDispatcher = KaliumDispatcherImpl,
+    private val scope: CoroutineScope
 ) {
 
     suspend operator fun invoke(
         conversationId: ConversationId,
         text: String,
         mentions: List<MessageMention> = emptyList(),
         quotedMessageId: String? = null
-    ): Either<CoreFailure, Unit> = withContext(dispatchers.io) {
+    ): Deferred<Either<CoreFailure, Unit>> = scope.async(dispatchers.io) {
```

Comment:
I wonder if there's any benefit on returning `Deferred`.

Maybe we could keep returning `Either<CoreFailure, Unit>>` but ignore cancellations by using a `SupervisorJob`?

Something like:
```suggestion
suspend operator fun invoke(
        conversationId: ConversationId,
        text: String,
        mentions: List<MessageMention> = emptyList(),
        quotedMessageId: String? = null
    ): Either<CoreFailure, Unit> = scope.async(dispatchers.io) {
        // Send message normally
    }.await()
```

If the outer scope (UI Scope / `viewModelScope`) is cancelled, the `scope.async` will keep doing its job.

Reply:
you are totally right dunno why I missed that